### PR TITLE
New Feature LocalStorage

### PR DESCRIPTION
**Pull request by miguelesco**
## What's new

- Add a new file to store all the local storage functions.
- new feature the inputs in contact save the values.

### CODE REVIEW

File: modules/formValidation.js [hunk 1]
```diff
@@ -37,6 +43,10 @@ const onSubmit = (event) => {
 };
 
 function InitFormValidation() {
+  firstname.addEventListener('input', () => Storage.save('firstname', firstname.value));
+  lastname.addEventListener('input', () => Storage.save('lastname', lastname.value));
+  fullname.addEventListener('input', () => Storage.save('fullname', fullname.value));
+  textArea.addEventListener('input', () => Storage.save('message', textArea.value));
   email.addEventListener('input', () => validateEmail());
   form.addEventListener('submit', (event) => onSubmit(event));
 }
```
Comment (Davidosky007) [inline modules/formValidation.js]: * When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data. currently this isn't working please fix it as its part of the project requirements

File: modules/storage.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      const formInputs = JSON.parse(window.localStorage.getItem('data'));
+      Storage.data = formInputs;
+    }
+
+    Object.keys(Storage.DOMelements).forEach((element) => {
+      Storage.DOMelements[element].value = Storage.data[element];
+    });
+  },
+};
+
```
Comment (Davidosky007) [inline modules/storage.js:31]: * currently when  the form is filled error appears in the console please fix it 🙏 
Screenshot (113)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @miguelesco, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: modules/formValidation.js (hunk 1 above)
Comment (miguelesco) [reply modules/formValidation.js]: the problem was that we change the function name and we forgot the change it there

File: modules/storage.js (hunk 2 above)
Comment (Davidosky007) [reply modules/storage.js:31]: * This error still persists kindly fix it
* 
Screenshot (113)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
* When the user loads the page or revisits  let data in the local storage be auto-populated with, the values entered in the previous visit to the page, currently this doesn't happen please fix

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @miguelesco, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
* Please fix the errors stated in the comment section check that it works then submit for review 👍 
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by miguelesco** [comment]
@Davidosky007 before making the review please make the pull to the updates of the codes, you can see the in the browser with github pages is working perfectly and even that name of the functions are different.

**Comment by miguelesco** [comment]
@Davidosky007 please do the review with github pages is currently working perfect

**Comment by miguelesco** [comment]
image

image

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**